docs/DeveloperGuide.adoc

```diff
@@ -150,97 +185,132 @@ The `Storage` component,
 
-The following sequence diagram shows how the undo operation works:
 
-image::UndoSequenceDiagram.png[]
+The proposed implementation of this algorithm is to encapsulate the logic of the algorithm into a `Command` class under
+`Logic` component, possibly with the help of some auxiliary classes. The command is then invoked when the user key in the relevant
+command keyword (refer user guide).
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+When the scheduling algorithm has finished running, it will update the data in the `Schedule` objects to reflect the
+allocated time slots to the interviewees, which the changes will then be reflected automatically in the UI.
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+=== Import
+==== Implementation
+
+The import feature uses `CsvReader` in the Model to read the given .csv file and stores the data into the model.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+* `import interviewer <csv file>` stores the read data as a list of `Schedule` objects in the model.
+* `import interviewee <csv file>` stores the read data as a list of `Interviewee` objects in the model.
+* `CsvReader` makes use of `BufferedReader` to read data from the given file.
 
-image::UndoRedoState4.png[]
+image::ImportSequenceDiagram.png[]
+Given above is an example of a sequence diagram for importing interviewer's schedules. It applies to both importing interviewee's and interviewer's data.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+As seen in the above sequence diagram, the execution of the import feature consists of these steps:
 
-image::UndoRedoState5.png[]
+* Step 1: `LogicManager` will start executing the command by parsing it over to `AddressBookParser`.
```

Comment:
Remember to replace the usage of AddressBook references with your own! :stuck_out_tongue_closed_eyes: